src/audio/codec_adapter/codec_adapter.c

```diff
@@ -428,6 +428,7 @@ int codec_adapter_copy(struct comp_dev *dev)
 		/* skipping as lib has not produced anything */
 		comp_err(dev, "codec_adapter_copy() error %x: lib hasn't processed anything",
 			 ret);
+		comp_update_buffer_consume(source, codec->cpd.consumed);
```

Comment:
I ..kind of... understand, from commit message, the reason for removing the frame - to not be decoded over and over again, but doesn't that mean that the frame is skipped? That frame is never processed, right?
Does it sound good playing an mp3, with this fix?

Are you sure the mp3 file is correct? Maybe the file, or more exactly that frame from the file, is corrupted.

Reply:
@iuliana-prodan Maybe one frame holds some metadata (ID3 or shit) instead of regular audio samples. It's valid but produces no real output (or too little and the rest needs to be accumulated in the internal state). So I wouldn't consider it impossible for this fix to be required.

I would guess that the "consumed" value is always set by the adapter, right?

Reply:
@iuliana-prodan this files are weird and come from QA. It looks like thise files are multiple files stitched together. So at the stitch point there is a weird frame. 
I have sent an email to Cadence support to let me know if this all right.

Reply:
So we need a comment here and probably a quirk i.e.
```c
 if (codec == cadence_mp3) then consume_frames();
```
OR better, this quirk should go in the Caence wrapper code rather than the generic code. @mrajwa any comments ?

Reply:
@lgirdwood I'm working with cadence to understand what happens. I do think that if this is an abornmal condition then we should receive an error, otherwise this is a valid situation and we should consume the frames no matter of codec family.

So, I think the patch should be applied as it is. Will get back to you after we elucidate the mistery.

Reply:
Can we at least have a comment then describing the issue, that way it will be easy to follow.

Reply:
@lgirdwood sure, will add the comment!

Reply:
ping, we can get this in rc3 if done today or tomorrow morning ?